versioned/tiered/src/main/java/com/dremio/nessie/versioned/impl/EntityType.java

```diff
@@ -0,0 +1,165 @@
+/*
+ * Copyright (C) 2020 Dremio
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.dremio.nessie.versioned.impl;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.function.Consumer;
+import java.util.function.Supplier;
+
+import com.dremio.nessie.tiered.builder.BaseConsumer;
+import com.dremio.nessie.tiered.builder.CommitMetadataConsumer;
+import com.dremio.nessie.tiered.builder.FragmentConsumer;
+import com.dremio.nessie.tiered.builder.L1Consumer;
+import com.dremio.nessie.tiered.builder.L2Consumer;
+import com.dremio.nessie.tiered.builder.L3Consumer;
+import com.dremio.nessie.tiered.builder.RefConsumer;
+import com.dremio.nessie.tiered.builder.ValueConsumer;
+import com.dremio.nessie.versioned.impl.PersistentBase.EntityBuilder;
+import com.dremio.nessie.versioned.store.Id;
+import com.dremio.nessie.versioned.store.NotFoundException;
+import com.dremio.nessie.versioned.store.SaveOp;
+import com.dremio.nessie.versioned.store.Store;
+import com.dremio.nessie.versioned.store.StoreOperationException;
+import com.dremio.nessie.versioned.store.ValueType;
+
+final class EntityType<C extends BaseConsumer<C>, E extends PersistentBase<C>> {
```

Comment:
Is "C" really the most appropriate letter to use here? It is being used for both Consumer and Supplier specializations of generic classes. Wouldn't T be less confusing?